lib/board.ex

```diff
@@ -66,28 +49,96 @@ defmodule Board do
     end
   end
 
+  def board_full?(board) do
+    board
+    |> Enum.reject(&mark?/1)
+    |> Enum.empty?
+  end
+
+  def dimension(board) do
+    length(board)
+    |> :math.sqrt
+    |> round
+  end
+
+  def current_lines(board) do
+    [rows(board), columns(board), diagonals(board)]
+    |> List.flatten
+    |> Enum.chunk(dimension(board))
+  end
+
+  def rows(board), do: Enum.chunk(board, dimension(board))
+
+  defp columns(board) do
+    board
+    |> rows
+    |> transpose
+  end
+
+  defp diagonals(board) do
+    [
+      backward_diagonal([], board, dimension(board), dimension(board) - 1),
```

Comment:
It's the same algorithm that I used in my Ruby Tic Tac Toe... I tried to extract some more methods which makes it a bit more readable.